### PR TITLE
Add Dark Engine

### PR DESCRIPTION
**Pull request by MichaelHinrichs**
### SteamDB app page links to a few games using this
https://steamdb.info/app/211600/
https://steamdb.info/app/211740/

### Brief explanation of the change
Add the Dark Engine, used in the Thief series by Looking Glass Studios.

### CODE REVIEW

**Comment by MichaelHinrichs** [comment]
Two apps.

**Comment by Lifeismana** [comment]
good news, that's what i marked 
image

File: rules.ini [hunk 1]
```diff
@@ -61,6 +61,7 @@ CryEngine[] = (?:^|/)CryD3DCompilerStub\.dll$
 CryEngine[] = (?:^|/)CryRenderD3D1[12]\.dll$
 CryEngine[] = (?:^|/)CryRenderVulkan\.dll$
 Danmakufu = (?:^|/)th_dnh\.def$
+DarkEngine = (?:^|/)DROMED\.ZIP$
```
Comment (Lifeismana) [inline rules.ini:64]:
```suggestion
DarkEngine = (?:^|/)res/intrface\.crf$
```

File: tests/types/Engine.DarkEngine.txt
Comment (Lifeismana) [inline tests/types/Engine.DarkEngine.txt:1]: add this string before running the script
```
Data/res/INTRFACE.CRF
```

**Comment by MichaelHinrichs** [comment]
> image
> 
> there's also another game that is supposed to use this engine, but idk if 3 games is enough

Actually, the Dark engine and Shock engine are supposed to be separate things, but they are very similar.

**Comment by MichaelHinrichs** [comment]
@Lifeismana Wait, so did you want me to ADD "res/intrface.crf", or to REPLACE "DROMED.ZIP" with it? Because you said to do one thing in "Engine.DarkEngine.txt", and the other in "rules.ini".

**Comment by Lifeismana** [comment]
Replace the rule and replace the old strings with the string i gave and then run the generate strings script

**Comment by SteamTracker** [comment]
After running changed rules in this pull request *(this bot only tests rules.ini changes)*:

Changed technologies: **Engine.DarkEngine**

These apps now match: **(3)** [211600](https://steamdb.info/app/211600/depots/) [211740](https://steamdb.info/app/211740/depots/) [238210](https://steamdb.info/app/238210/depots/)

Please confirm that all these changed apps are correct.

**Comment by Lifeismana** [comment]
not used by enough apps

**Pull request closed without merging**